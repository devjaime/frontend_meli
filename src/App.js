import React from 'react';
import './styles/App.scss';
import { Route, Switch } from 'react-router-dom';
import Home from './components/Home';
import Results from './components/Results';
import Product from './components/Product';

function App() {
  return (
    <div>
      <Route component={Home} path='/' />
      <Switch>
        <Route component={Results} exact path='/items' />
        <Route component={Product} path='/items/:id' />
      </Switch>
    </div>
  );
}

export default App;
