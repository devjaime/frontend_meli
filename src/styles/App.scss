// Keep all styles in same sheet to allow components to access to them everywhere
@import './variables';
@import './searchBox';
@import './results';
@import './product';

html, body {
    background-color: $meli-light-grey;
}