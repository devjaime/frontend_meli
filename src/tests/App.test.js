import React from 'react';
import { shallow } from "enzyme";
import {App} from '../../src/App';
import { mount } from "enzyme";
import { MemoryRouter, Route } from "react-router-dom";

describe('Test in <App/>', () => {
    test('should display the component correctly', () => {
        const wrapper =  shallow( <App/>)
        expect(wrapper).toMatchSnapshot();
    })

    test('should display the component correctly', () => {
        let testHistory;
        let testLocation;
        let History={"length":2,"action":"PUSH","location":{"pathname":"/items","search":"?search=monitor curvo","hash":"","key":"q06uc0"}};
        let Location={"pathname":"/items","search":"?search=monitor curvo","hash":"","key":"q06uc0"};
        const wrapper = mount(
            <MemoryRouter initialEntries={[`/`]}>
            <App />
            <Route
                path={`*`}
                render={routeProps => {
                testHistory = History
                testLocation = Location
                return null
                }}/>
            </MemoryRouter>
        );
        expect(wrapper).toMatchSnapshot();
        expect(testHistory).toMatchSnapshot();
        expect(testLocation).toMatchSnapshot();
    })
})