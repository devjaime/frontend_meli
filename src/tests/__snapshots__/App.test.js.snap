// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <App/> should display the component correctly 1`] = `
<div>
  <Route
    component={[Function]}
    path="/"
  />
  <Switch>
    <Route
      component={[Function]}
      exact={true}
      path="/items"
    />
    <Route
      component={[Function]}
      path="/items/:id"
    />
  </Switch>
</div>
`;

exports[`Test in <App/> should display the component correctly 2`] = `
Array [
  <div>
    <header
      className="header"
    >
      <div>
        <div
          onClick={[Function]}
          onKeyDown={[Function]}
          role="button"
          tabIndex={0}
        >
          <img
            alt="logo_ml"
            className="meli-logo"
            src="logo_ml.png"
            title="Mercado Libre - Donde comprar y vender de todo"
          />
        </div>
        <form
          onSubmit={[Function]}
        >
          <input
            id="serach_input"
            name="search"
            onChange={[Function]}
            placeholder="Buscar productos, marcas y mÃ¡s"
            type="text"
            value=""
          />
          <button
            type="submit"
          >
            <img
              alt="ic_search"
              src="ic_search.png"
              title="Buscar"
            />
          </button>
        </form>
      </div>
    </header>
  </div>,
  "",
]
`;

exports[`Test in <App/> should display the component correctly 3`] = `
Object {
  "action": "PUSH",
  "length": 2,
  "location": Object {
    "hash": "",
    "key": "q06uc0",
    "pathname": "/items",
    "search": "?search=monitor curvo",
  },
}
`;

exports[`Test in <App/> should display the component correctly 4`] = `
Object {
  "hash": "",
  "key": "q06uc0",
  "pathname": "/items",
  "search": "?search=monitor curvo",
}
`;
